lib/rouge/lexers/ghc_cmm.rb

```diff
@@ -0,0 +1,337 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+# C minus minus (Cmm) is a pun on the name C++. It's an intermediate language
+# of the Glasgow Haskell Compiler (GHC) that is very similar to C, but with
+# many features missing and some special constructs.
+#
+# Cmm is a dialect of C--. The goal of this lexer is to use what GHC produces
+# and parses (Cmm); C-- itself is not supported.
+#
+# https://gitlab.haskell.org/ghc/ghc/wikis/commentary/compiler/cmm-syntax
+#
+module Rouge
+  module Lexers
+    class GHCCmm < RegexLexer
+      title "GHC Cmm (C--)"
+      desc "GHC Cmm is the intermediate representation of the GHC Haskell compiler"
+      tag 'ghc-cmm'
+      aliases 'cmm'
+      filenames '*.cmm', '*.dump-cmm', '*.dump-cmm-*'
+
+      macros = %r(#(?:define|include|endif|else|if))
+
+      ws = %r(\s|//.*?\n|/[*](?:[^*]|(?:[*][^/]))*[*]+/)mx
+      id = %r((?!#{macros})[\w#\$%_']+)
```

Comment:
Do we need to interpolate the macros here? Would `\b` not suffice?

Reply:
Well, it looks like it can be even simpler. :smiley: 

(Please see change.)